src/components/SignUp/SignUp.js

```diff
@@ -0,0 +1,127 @@
+import { Link, Navigate } from "react-router-dom";
+import { useState } from "react";
+
+import UserService from "../../services/userService";
+import Navbar from "components/Navbar/Navbar";
+import FormInput from "components/FormInput/FormInput";
+import FormSelect from "components/FormSelect/FormSelect";
+import LandingVideo from "components/LandingVideo/LandingVideo";
+
+import "./SignUp.scss";
+
+const SignUp = () => {
+  const GENDER_OPTIONS = ["male", "female"];
+
+  const [name, setName] = useState("");
+  const [email, setEmail] = useState("");
+  const [password, setPassword] = useState("");
+  const [passwordConfirmation, setPasswordConfirmation] = useState("");
+  const [gender, setGender] = useState("");
+  const [redirect, setRedirect] = useState(false);
+
+  async function handleSubmit(event) {
+    event.preventDefault();
+
+    if(!event.target.checkValidity()) return;
+
+    const body = {
+      user: {
+        first_name: name,
+        email: email,
+        gender: gender,
+        password: password,
+        password_confirmation: passwordConfirmation
+      }
+    };
+
+
+    try {
+      await UserService.signUp(body);
+      setRedirect(true);
+    } catch (error) {
+      const { response: {data: {errors}} } = error;
+      alert(errors.full_messages[0]);
+    }
+  };
+
+  if (redirect) {
+    return <Navigate to="/" />
+  }else{
+
+    return (
+      <div className="sign-up">
+        <section className="container">
+          <Navbar className="white" />
+  
+          <h1 id="sign-up-title">Sign Up</h1>
+  
+          <form className="sign-up-form" onSubmit={handleSubmit}>
+            <FormInput
+              type="text"
+              id="name"
+              name="name"
+              label="name"
+              value={name}
+              onChange={(value) => setName(value)}
+              required={true}
+            />
+  
+            <FormInput
+              type="email"
+              id="email"
+              name="email"
+              label="email"
+              value={email}
+              onChange={(value) => setEmail(value)}
```

Comment:
You could make it easy only sending setName in onChange prop

```suggestion
              onChange={setEmail}
```